policies/templates/gcp_bq_dataset_location_v1.yaml

```diff
@@ -0,0 +1,182 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This template is for policies restricting the locations 
+# of BigQuery datasets to specific locations in GCP. It supports 
+# allowlist or denylist modes, as well as exempting selected 
+# datasets from the list.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-bq-dataset-location-v1
```

Comment:
nit: "gcp-bigquery" to be consistent with the kind below?

Reply:
```suggestion
  name: gcp-bigquery-dataset-location-v1
```